src/OpenColorIO/Config.cpp

```diff
@@ -2610,5 +2617,113 @@ ConstConfigRcPtr Config::Impl::Read(std::istream & istream, const char * filenam
     return config;
 }
 
+void Config::Impl::checkVersionConsistency(ConstTransformRcPtr & transform) const
```

Comment:
Just a question. Do you think we should add a version compatibility method to the `Transform` interface, so each transform can determine its own compatibility outside of `Config`? I don't have a problem with this implementation; was just thinking that this is the sort of function that can balloon over time.

Reply:
That's a valid point.
The rational is to avoid having a version related method in the public API (i.e. not useful for any public usage) and to centralize all the checks. There is also the point that it means some code for the first implementation but it should not change a lot in the future.